contracts/Core/BlockManager.sol

```diff
@@ -224,14 +224,21 @@ contract BlockManager is Initializable, BlockStorage, StateManager, BlockManager
             return;
         }
 
-        for (uint8 i = 0; i < sortedProposedBlockslength; i++) {
-            // Replace : New Block has better biggest influence
-            if (proposedBlocks[epoch][sortedProposedBlockIds[epoch][i]].biggestInfluence < biggestInfluence) {
-                sortedProposedBlockIds[epoch][i] = blockId;
-                return;
+        if (proposedBlocks[epoch][sortedProposedBlockIds[epoch][0]].biggestInfluence > biggestInfluence) {
+            return;
```

Comment:
at line 71, numProposed blocks is increased even though the block was not inserted.

Reply:
Won't that be the case even if a valid block is proposed but length is greater than maxAltBlocks?

Reply:
no. in that case a block is actually inserted.

Reply:
But a block is popped off as we will have to maintain the length of maxAltBlocks and numProposedBlocks is also being used as ID for each block that has been proposed.

Reply:
why increase numProposed if the block was never inserted. Its as if the block was never proposed.

Reply:
Fixed https://github.com/razor-network/contracts/pull/602/commits/4e178a31d50dd430d69a67a545da1339b41663c7